GoogleMapsComponents/Maps/BaseListableEntity.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.JSInterop;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GoogleMapsComponents.Maps
+{
+    public abstract class BaseListableEntity<T> : IDisposable, IJsObjectRef
+        where T: BaseListableEntityOptions
+    {
+        protected readonly JsObjectRef _jsObjectRef;
+
+        public readonly Dictionary<string, List<MapEventListener>> EventListeners;
+
+        public Guid Guid => _jsObjectRef.Guid;
+
+        internal BaseListableEntity(JsObjectRef jsObjectRef)
+        {
+            _jsObjectRef = jsObjectRef;
+            EventListeners = new Dictionary<string, List<MapEventListener>>();
+        }
+
+        public void Dispose()
+        {
+            foreach (string key in EventListeners.Keys)
+            {
+                //Probably superfluous...
+                if (EventListeners[key] != null)
```

Comment:
Two request to key. Better use  if((EventListeners.TryGetValue(key, out var eventsList) && eventsList != null))

Reply:
True, my fault, I'm used to the "old school".